plugins/inputs/dpdk/dpdk.go

```diff
@@ -0,0 +1,263 @@
+  ## The list of available commands depend on the application deployed. Applications can register their own commands
+  ##   via telemetry library API http://doc.dpdk.org/guides/prog_guide/telemetry_lib.html#registering-commands
+  ## For e.g. L3 Forwarding with Power Management Sample Application this could be: 
+  ##   additional_commands = ["/l3fwd-power/stats"]
+  # additional_commands = []
+
+  ## Allows turning off collecting data for individual "ethdev" commands.
+  ## Remove "/ethdev/link_status" from list to start getting link status metrics.
+  [inputs.dpdk.ethdev]
+    exclude_commands = ["/ethdev/link_status"]
+
+  ## When running multiple instances of the plugin it's recommended to add a unique tag to each instance to identify
+  ## metrics exposed by an instance of DPDK application. This is useful when multiple DPDK apps run on a single host.  
+  ##  [inputs.dpdk.tags]
+  ##    dpdk_instance = "my-fwd-app"
+`
+	defaultPathToSocket        = "/var/run/dpdk/rte/dpdk_telemetry.v2"
+	defaultAccessTimeout       = config.Duration(200 * time.Millisecond)
+	maxCommandLength           = 56
+	maxCommandLengthWithParams = 1024
+	pluginName                 = "dpdk"
+	ethdevListCommand          = "/ethdev/list"
+	rawdevListCommand          = "/rawdev/list"
+)
+
+type dpdk struct {
+	SocketPath         string          `toml:"socket_path"`
+	AccessTimeout      config.Duration `toml:"socket_access_timeout"`
+	DeviceTypes        []string        `toml:"device_types"`
+	EthdevConfig       ethdevConfig    `toml:"ethdev"`
+	AdditionalCommands []string        `toml:"additional_commands"`
+	Log                telegraf.Logger `toml:"-"`
+
+	connector                    *dpdkConnector
+	rawdevCommands               []string
+	ethdevCommands               []string
+	ethdevExcludedCommandsFilter filter.Filter
+}
+
+type ethdevConfig struct {
+	EthdevExcludeCommands []string `toml:"exclude_commands"`
+}
+
+func init() {
+	inputs.Add(pluginName, func() telegraf.Input {
+		dpdk := &dpdk{
+			// Setting it here (rather than in `Init()`) to distinguish between "zero" value,
+			// default value and don't having value in config at all.
+			AccessTimeout: defaultAccessTimeout,
+		}
+		return dpdk
+	})
+}
+
+func (dpdk *dpdk) SampleConfig() string {
+	return sampleConfig
+}
+
+func (dpdk *dpdk) Description() string {
+	return description
+}
+
+// Performs validation of all parameters from configuration
+func (dpdk *dpdk) Init() error {
+	if dpdk.SocketPath == "" {
+		dpdk.SocketPath = defaultPathToSocket
+		dpdk.Log.Debugf("using default '%v' path for socket_path", defaultPathToSocket)
+	}
+
+	if dpdk.DeviceTypes == nil {
+		dpdk.DeviceTypes = []string{"ethdev"}
+	}
+
+	var err error
+	if err = isSocket(dpdk.SocketPath); err != nil {
+		return err
+	}
+
+	dpdk.rawdevCommands = []string{"/rawdev/xstats"}
+	dpdk.ethdevCommands = []string{"/ethdev/stats", "/ethdev/xstats", "/ethdev/link_status"}
+
+	if err = dpdk.validateCommands(); err != nil {
+		return err
+	}
```

Comment:
You can use local errors here
```suggestion
	if err := isSocket(dpdk.SocketPath); err != nil {
		return err
	}

	dpdk.rawdevCommands = []string{"/rawdev/xstats"}
	dpdk.ethdevCommands = []string{"/ethdev/stats", "/ethdev/xstats", "/ethdev/link_status"}

	if err := dpdk.validateCommands(); err != nil {
		return err
	}
```
but as it is only cosmetic, take this as FYI only. :-)